pkg/stream/pipeline.go

```diff
@@ -83,7 +83,7 @@ func (p *Pipeline) BootWithInterfaces(logger mon.Logger, metric mon.MetricWriter
 
 func (p *Pipeline) Run(ctx context.Context) error {
 	defer p.logger.Info("leaving pipeline")
```

Comment:
again i believe, this log should be the last thing that is done within this function